spec.bs

```diff
@@ -385,7 +385,9 @@ dictionary AuctionAdConfig {
   USVString directFromSellerSignals;
   unsigned long long sellerTimeout;
   unsigned short sellerExperimentGroupId;
-  record<USVString, any> perBuyerSignals;
+
+  // Expects a record<USVString, any> or a Promise resolving to that.
+  any perBuyerSignals;

```

Comment:
Wait why can't this be a `Promise<record<USVString, any>>`?

Reply:
Ah I misread this. I guess what I meant (if I were reading it more closely) would be: can this be `CustomType` where `CustomType` is `typedef (record<USVString, any> or Promise<record<USVString, any>>) CustomType;`

Reply:
But that doesn't solve the Promise-can't-participate-in-union-types problem, right? WDYT about that Jeffrey? I think here we want to accept a record **or** a Promise that resolves to a record.

Reply:
Accepting `Promise<record<...>` does accept a record or a Promise that resolves to a record.

Reply:
Note that if we go that way, it will have an observable behavior difference with what's currently implemented: we currently throw an exception immediately at runAdAuction time if such a maybe-promise field has invalid data as non-promise, while if you use WebIDL with Promise<record<...>> I think you need to have microtasks to run, then eventually the promise returned from runAdAuction gets rejected.

... There may also be difference in how thenable things are handled, though, and we are probably wrong to not treat them as promises in our impl?

(There is also an exception thrown within the .then of the synthesized wrapper promise, not sure if /that/ is observable).

I am somewhat reluctant to basically do all the extra work of auto-wrapping in promises than doing a bunch of further asynchronous processing when not needed (and getting worse debugability for that), but maybe that's what's expected?

Reply:
OK, switched to that, and added a TODO entry for me to change the implementation to ask. Hopefully people won't Hyrum's law themselves into actually relying on the differences...

Now this is much simplified, since it's basically just promise glue and a helper.  If this is OK, I'll do it for rest of the fields, and that hopefully should be reasonably readable...